pwa-devdocs/src/pagebuilder/custom-components/add-component/index.md

```diff
@@ -0,0 +1,147 @@
+
+```jsx
+const toHTML = str => ({ __html: str });
+
+return (
+        <div>
+            <div>{quote}</div>
+            <div>{author}</div>
+            <div dangerouslySetInnerHTML={toHTML(description)} />
+        </div>
+    );
+```
+
+Because our `description` property value is HTML, we need to use React's [dangerouslySetInnerHTML] and `__html` to render it out properly.
+
+## Step 3: Apply form styles
+
+To make it easy to apply all the inline styles from your content type to your Page Builder component, you can create a separate object that contains only the styles retrieved from the content type's form. These styles include those that are retrieved by [utility functions] like `getAdvanced()` and `getBackgroundImages()`.
+
+For our component, we created an object called `formStyles` to group our form styles as shown here:
+
+```js
+const formStyles = {
+    textAlign,
+    border,
+    borderColor,
+    borderWidth,
+    borderRadius,
+    isHidden,
+    marginTop,
+    marginRight,
+    marginBottom,
+    marginLeft,
+    paddingTop,
+    paddingRight,
+    paddingBottom,
+    paddingLeft
+};
+```
+
+These are the styles returned from the Advanced form section of our Quote content type (using `getAdvanced()`). When they are grouped together as shown, we can apply them all at once to our component, as follows:
+
+```jsx
+return (
+    <div style={formStyles}>
+        <div>{quote}</div>
+        <div>{author}</div>
+        <div dangerouslySetInnerHTML={toHTML(description)} />
+    </div>
+);
+```
+
+Now we need to apply the CSS classes we used to style the rest of our content type.
+
+## Step 4: Apply CSS classes
+
+Applying CSS classes to your component works the same as apply these classes to any other React component: first import your CSS stylesheet, then add the stylesheet classes to your JSX in the usual way, using `className`.
+
+For our component, we import our component stylesheet:
+
+```js
+import quoteStyles from './exampleQuote.css';
+```
+
+Then apply the classes within to the correct nodes:
+
+```jsx
+return (
+    <div style={formStyles}>
+        <div className={quoteStyles.quote}>{quote}</div>
+        <div className={quoteStyles.quoteAuthor}>{author}</div>
+        <div
+            className={quoteStyles.quoteDescription}
+            dangerouslySetInnerHTML={toHTML(description)}
+        />
+    </div>
+);
+```
+
+In this code, we only used one class per node. However, we often need to apply several classes to a node. To do this, you need to create a variable that concatenates your classes into a single string.
+
+For example, the `qoute` node (first child) of our content type requires two classes to reproduce it faithfully in our component: `.quote` and `.blueQuote`. To solve this, we created another variable call `quoteClassName` and used it to add both classes to the `quote` node as follows:
+
+```jsx
+const quoteClassName = [quoteClasses.quote, quoteClasses.blueQuote].join(' ');
```

Comment:
We should avoid merging multiple classes together like this, instead we should compose classes together in the CSS style.

You can see an example of that here: https://github.com/magento/pwa-studio/blob/feature/page-builder/packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Banner/banner.css#L41

Let me know if you have questions